src/metadata.rs

```diff
@@ -0,0 +1,92 @@
+use crate::LiveFile;
+use crate::{ffi, ffi_util::from_cstr};
+use std::ffi::CString;
+use std::fmt::Debug;
+use std::ptr;
+
+/// Metadata for column family import/export.
+#[derive(Debug)]
+pub struct ExportImportFilesMetaData {
+    pub(crate) inner: *mut ffi::rocksdb_export_import_files_metadata_t,
+}
+
+impl ExportImportFilesMetaData {
+    pub fn get_db_comparator_name(&self) -> String {
+        unsafe {
+            let c_name =
+                ffi::rocksdb_export_import_files_metadata_get_db_comparator_name(self.inner);
+            from_cstr(c_name)
+        }
+    }
+
+    pub fn set_db_comparator_name(&mut self, name: &str) {
+        let c_name = CString::new(name.as_bytes()).unwrap();
+        unsafe {
+            ffi::rocksdb_export_import_files_metadata_set_db_comparator_name(
+                self.inner,
+                c_name.as_ptr(),
+            );
+        };
+    }
+
+    pub fn get_files(&self) -> Vec<LiveFile> {
+        unsafe {
+            let livefiles_ptr = ffi::rocksdb_export_import_files_metadata_get_files(self.inner);
+            let files = LiveFile::from_rocksdb_livefiles_ptr(livefiles_ptr);
+            ffi::rocksdb_livefiles_destroy(livefiles_ptr);
+            files
+        }
+    }
+
+    pub fn set_files(&mut self, files: &Vec<LiveFile>) {
+        unsafe {
+            let livefiles = ffi::rocksdb_livefiles_create();
+
+            for file in files {
+                let c_cf_name = CString::new(file.column_family_name.clone()).unwrap();
+                let c_name = CString::new(file.name.clone()).unwrap();
+                let c_directory = CString::new(file.directory.clone()).unwrap();
+
+                ffi::rocksdb_livefiles_add(
+                    livefiles,
+                    c_cf_name.as_ptr(),
+                    c_name.as_ptr(),
+                    c_directory.as_ptr(),
+                    file.size,
+                    file.level,
+                    file.start_key
+                        .as_ref()
+                        .map_or(ptr::null(), |k| k.as_ptr() as *const i8),
+                    file.start_key.as_ref().map_or(0, Vec::len),
+                    file.end_key
+                        .as_ref()
+                        .map_or(ptr::null(), |k| k.as_ptr() as *const i8),
+                    file.end_key.as_ref().map_or(0, Vec::len),
+                    file.smallest_seqno,
+                    file.largest_seqno,
+                    file.num_entries,
+                    file.num_deletions,
+                );
+            }
+
+            ffi::rocksdb_export_import_files_metadata_set_files(self.inner, livefiles);
+            ffi::rocksdb_livefiles_destroy(livefiles);
+        }
+    }
```

Comment:
I guess one way to avoid the back and forth from C to Rust to C (in order to reuse the `LiveFile`) could be to also store `ExportImportFilesMetaData` as a pure Rust struct. The underlying C++ struct seems to contain a `List<LiveFileMetadataData>` and a `std::string db_comparator`. Then, however, one needs to create the `ffi::rocksdb_export_import_files_metadata_t` when one calls `create_column_family_with_import`. 

Given that this is not a frequent operation, both ways are probably fine.

Reply:
Yeah, that was exactly my initial approach. I got a bit tangled up with trying to pass in an entire array of file data into C++, I might give this another shot now that I understand what (not) to do :-)

Reply:
The risk in the roundtrip conversion is that we might lose data if new fields were added in rocksdb.

Reply:
That's precisely what I was hoping to avoid. But consider that whoever uses this API will probably need to somehow communicate the metadata represented in `LiveFile` for import, most likely going through a serde roundtrip. If the Rust struct doesn't fully capture the underlying C++ object, we risk losing data anyway. I don't have a good way to solve that.

Reply:
Updated this! The many-parameter signature is going to be a pain in the long run so I rewrote this to use a `rocksdb_livefile_t` wrapper struct. Unfortunately I still had to expose individual setters so the amount of code on the Rust side is basically unchanged.